variables.tf

```diff
@@ -100,13 +100,44 @@ variable "resource_names_map" {
 variable "vpc_id" {
   description = "The VPC ID of the VPC where infrastructure will be provisioned"
   type        = string
+  default     = null
+}
+
+variable "vpc_name" {
```

Comment:
A suggestion: instead of having multiple variables for vpc details, you can have an object with fields vpc_name, subnet_cidr, subnet_names etc.